pixi.toml

```diff
@@ -60,9 +60,10 @@ clean = { cmd = "pip uninstall -y bodo bodosql bodo-iceberg-connector && bash bu
 
 
 [system-requirements]
-macos = "11.0"
+macos = "12.0"
 
 [dependencies]
+python = "3.12.*"
```

Comment:
Why do we need this?

Reply:
I realized after we made some other changes in Pixi, we never locked the Python version for the platform and dev. Its ok right now because Numba supports 3.12 at most, but when they support 3.13, we dont want it to auto-upgrade.

Note, Pixi doesn't automatically take the info from the `pyproject.toml`. They do have an option to write it in the `pyproject.toml` file, but I ran into some issues and opened tickets on their side.